haddock/workflow.py

```diff
@@ -0,0 +1,98 @@
+"""HADDOCK3 workflow logic"""
+import logging
+import importlib
+import shutil
+from pathlib import Path
+import toml
+from haddock.error import HaddockError, RecipeError
+from haddock.defaults import MODULE_PATH_NAME, TOPOLOGY_PATH
+
+
+logger = logging.getLogger(__name__)
+
+
+class WorkflowManager:
+    """The WorkflowManager reads the recipes and executes them."""
+    def __init__(self, recipe_path, start=0):
+        path = Path(recipe_path)
+        if not path.exists():
+            raise HaddockError(f"Recipe {recipe_path} does not exist or cannot"
+                               " be opened")
+        self.start = start
+        # Save current working path
+        self.workspace = path.parent.absolute()
+        # Create a recipe from a TOML file
+        self.recipe = Recipe(path)
+
+    def run(self):
+        """High level workflow composer"""
+        for step in self.recipe.steps[self.start:]:
+            step.execute()
+
+
+class Recipe:
```

Comment:
If not using metaphors anymore, maybe there is a better name for this class too?